Source/Services/LIFEDataProvider.m

```diff
@@ -75,6 +76,58 @@ - (instancetype)init
     LIFE_THROW_UNAVAILABLE_EXCEPTION(initWithReportOwner:SDKVersion:);
 }
 
+#pragma mark - Client Events
+
+- (void)logClientEventWithName:(nonnull NSString *)eventName afterDelay:(NSTimeInterval)delay
+{
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delay * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        [self logClientEventWithName:eventName];
+    });
+}
+
+- (void)logClientEventWithName:(nonnull NSString *)eventName
+{
+    LIFELogIntDebug(@"Logging event: \"%@\"", eventName);
+    
+    NSMutableDictionary *params = [NSMutableDictionary dictionary];
+    
+    [self.reportOwner switchCaseAPIKey:^(NSString *apiKey) {
+        [LIFENSMutableDictionaryify(params) life_safeSetObject:apiKey forKey:@"api_key"];
+    } email:^(NSString * _Nonnull email) {
+        [LIFENSMutableDictionaryify(params) life_safeSetObject:email forKey:@"email"];
+    }];
+
+    NSString *deviceIdentifier = [UIDevice currentDevice].identifierForVendor.UUIDString;
+    NSString *userIdentifier = [Buglife sharedBuglife].userIdentifier;
+    NSString *userEmail = [Buglife sharedBuglife].userEmail;
+    NSMutableDictionary *clientEventParams = [NSMutableDictionary dictionary];
+    [clientEventParams life_safeSetObject:_sdkVersion forKey:@"sdk_version"];
```

Comment:
Was going to say "We're logging all this on every event? Seems excessive," but I didn't realize this is sending stats back to Buglife HQ immediately. Maybe we should have a better method name then?